graylog2-web-interface/src/stores/indices/IndexSetsStore.jsx

```diff
@@ -128,6 +128,26 @@ const IndexSetsStore = Reflux.createStore({
     IndexSetsActions.setDefault.promise(promise);
   },
 
+  stats() {
+    const url = URLUtils.qualifyUrl(ApiRoutes.IndexSetsApiController.stats().url);
+    const promise = fetch('GET', url);
```

Comment:
If this is a long-running operation (which it probably becomes when the number of indices is big enough), we should cache the promise and return it if it still running instead of creating a new one. The general logic for this would look like this:

```
if (this.promise) {
  return this.promise;
}
this.promise = fetch(..);
this.promise.then(...).finally(() => { this.promise = undefined; });
```

This pattern avoids building up too many requests in the browser.